### PR TITLE
gh #21 L1 TCs for dsSetAudioMixerLevels API

### PR DESCRIPTION
**Pull request by krishna-1809**

### CODE REVIEW

File: src/test_l1_dsAudio.c [hunk 1]
```diff
@@ -9954,6 +9954,117 @@ void test_l1_dsAudio_negative_dsGetHDMIARCPortId(void) {
...
+
+	// Step 02: Get the port handle for all supported audio ports
+	for (int i = 0; i < NUM_OF_PORTS; i++) {
+		result = dsGetAudioPort(kPorts[i].id.type, kPorts[i].id.index, &handle[i]);
+		DS_ASSERT_AUTO_TERM_NUMERICAL(result, dsERR_NONE);
+		DS_ASSERT_AUTO_TERM_NOT_EQUAL(handle[i], null_handle);
+
+		// Step 03: Call dsSetAudioMixerLevels() for each port by looping through dsAudioInput_t enum
+		for (dsAudioInput_t audioInput = dsAudioInputPrimary; audioInput < dsAudioInputMax; audioInput++) {
+			DS_ASSERT_AUTO_TERM_NUMERICAL(dsSetAudioMixerLevels(handle[i], audioInput, volume), dsERR_NONE);
```
Comment (Ulrond) [inline src/test_l1_dsAudio.c:9995]: please break this up to use UT_ASSERT Functions correctly, and not macro the offical macros

File: src/test_l1_dsAudio.c [hunk 2]
```diff
@@ -9954,6 +9954,117 @@ void test_l1_dsAudio_negative_dsGetHDMIARCPortId(void) {
...
+	intptr_t  handle[NUM_OF_PORTS] = {INT_ARRAY_INIT};
+
+	// Step 01: Initialize Audio Port using dsAudioPortInit()
+	result = dsAudioPortInit();
+	UT_ASSERT_EQUAL(result, dsERR_NONE);
+
+	// Step 02: Get the port handle for all supported audio ports
+	for (int i = 0; i < NUM_OF_PORTS; i++) {
+		result = dsGetAudioPort(kPorts[i].id.type, kPorts[i].id.index, &handle[i]);
+		DS_ASSERT_AUTO_TERM_NUMERICAL(result, dsERR_NONE);
```
Comment (Ulrond) [inline src/test_l1_dsAudio.c:9990]: What is DS_ASSERT? 

You can only use UT_ASSERT funtions

File: src/test_l1_dsAudio.c (hunk 1 above)
Comment (Ulrond) [reply src/test_l1_dsAudio.c:9995]: how can you validate the that volume is set correct?

File: src/test_l1_dsAudio.c [hunk 3]
```diff
@@ -9954,6 +9954,117 @@ void test_l1_dsAudio_negative_dsGetHDMIARCPortId(void) {
...
+	// Step 04: Get the port handle for all supported audio ports
+	for (int i = 0; i < NUM_OF_PORTS; i++) {
+		result = dsGetAudioPort(kPorts[i].id.type, kPorts[i].id.index, &handle[i]);
+		DS_ASSERT_AUTO_TERM_NUMERICAL(result, dsERR_NONE);
+		DS_ASSERT_AUTO_TERM_NOT_EQUAL(handle[i], null_handle);
+
+		// Step 05: Set Audio Mixer level for Invalid Audio Input(dsAudioInputMax)
+		DS_ASSERT_AUTO_TERM_NUMERICAL(dsSetAudioMixerLevels(handle[i],dsAudioInputMax, volume), dsERR_INVALID_PARAM);
+
+		// Step 06: Set Audio Mixer level for Invalid Volume Level()
```
Comment (Ulrond) [inline src/test_l1_dsAudio.c:10055]: where is your volume < 0 & > 100 tests?

File: src/test_l1_dsAudio.c [hunk 4]
```diff
@@ -9954,6 +9954,117 @@ void test_l1_dsAudio_negative_dsGetHDMIARCPortId(void) {
...
+	intptr_t  handle[NUM_OF_PORTS] = {INT_ARRAY_INIT};
+
+	// Step 01: Call dsSetAudioMixerLevels() without prior initialization of Audio Port
+	UT_ASSERT_EQUAL(dsSetAudioMixerLevels(-1, dsAudioInputPrimary, 75), dsERR_NOT_INITIALIZED);
+
+	// Step 02: Initialize Audio Port using dsAudioPortInit()
+	UT_ASSERT_EQUAL(dsAudioPortInit(), dsERR_NONE);
+
+	// Step 03: Call dsSetAudioMixerLevels() with an invalid port handle
+	DS_ASSERT_AUTO_TERM_NUMERICAL(dsSetAudioMixerLevels(handle[0], dsAudioInputPrimary, 75), dsERR_INVALID_PARAM);
```
Comment (Ulrond) [inline src/test_l1_dsAudio.c:10044]: Use UT Macros correctly please , Terminate should be at the end of your function not during.


Remove all uses of ->

DS_ASSERT_AUTO_TERM_NUMERICAL -> all tests need to fall through. -> UT_ASSERT
DS_ASSERT_AUTO_TERM_NOT_EQUAL -> all tests need to fall through -> UT_ASSERT_NOT_EQUAL
DS_ASSERT_AUTO_TERM_STRING -> remove and fall through.  -> UT_ASSERT_STRING

**Review by Ulrond — CHANGES_REQUESTED** [review]
questions

**Comment by Ulrond** [comment]
can you run run the full test for dsAudio and post the results please.

You can ignore the new changes to dsAudio to give the initial results if you've not done the code changes yet.

File: src/test_l1_dsAudio.c (hunk 2 above)
Comment (krishna-1809) [reply src/test_l1_dsAudio.c]: Updated with UT_ASSERT functions for the entire document

File: src/test_l1_dsAudio.c (hunk 3 above)
Comment (krishna-1809) [reply src/test_l1_dsAudio.c]: Added negative volume level check too

File: src/test_l1_dsAudio.c (hunk 4 above)
Comment (krishna-1809) [reply src/test_l1_dsAudio.c]: Addressed

**Review by hari22yuva — CHANGES_REQUESTED** [review]
Please address the review comments and share the test report

File: src/test_l1_dsAudio.c [hunk 5]
```diff
@@ -9954,6 +9930,132 @@ void test_l1_dsAudio_negative_dsGetHDMIARCPortId(void) {
...
+
+	// Step 02: Get the port handle for all supported audio ports
+	for (int i = 0; i < NUM_OF_PORTS; i++) {
+		result = dsGetAudioPort(kPorts[i].id.type, kPorts[i].id.index, &handle[i]);
+		UT_ASSERT_EQUAL(result, dsERR_NONE);
+		UT_ASSERT_NOT_EQUAL(handle[i], null_handle);
+
+		// Step 03: Call dsSetAudioMixerLevels() for each port by looping through dsAudioInput_t enum
+		for (dsAudioInput_t audioInput = dsAudioInputPrimary; audioInput < dsAudioInputMax; audioInput++) {
+			result = dsSetAudioMixerLevels(handle[i], audioInput, volume);
```
[the remark this answers is not in the document]
Comment (krishna-1809) [reply src/test_l1_dsAudio.c:9971]: What parameters will be sent in this case ?

File: src/test_l1_dsAudio.c [hunk 6]
```diff
@@ -9954,6 +9930,132 @@ void test_l1_dsAudio_negative_dsGetHDMIARCPortId(void) {
...
+*
+* **Dependencies:** None@n
+* **User Interaction:** None
+*
+* **Test Procedure:**@n
+* |Variation / Step|Description|Test Data|Expected Result|Notes|
+* |:--:|-----------|----------|--------------|-----|
+* |01|Initialize Audio Ports using dsAudioPortInit() | | dsERR_NONE | Should Pass |
+* |02|Get the audio port handle for valid audio port type and valid index | type, index = [Loop through kPorts] , handle = [valid handle] | dsERR_NONE | Valid port handle must be returned |
+* |03|Call dsSetAudioMixerLevels() by looping through the primary audio inputs | handle = [valid handle], aInput = dsAudioInputPrimary , volume: 75 | dsERR_NONE | Should Pass |
```
[the remark this answers is not in the document]
Comment (krishna-1809) [reply src/test_l1_dsAudio.c]: Addressed

File: src/test_l1_dsAudio.c (hunk 5 above)
[the remark this answers is not in the document]
Comment (krishna-1809) [reply src/test_l1_dsAudio.c]: Addressed

**Review by Ulrond — APPROVED** [review]
looks good

**Review by hari22yuva — APPROVED** [review]
ship it

**Pull request merged**